### PR TITLE
Move webdb

### PR DESCRIPTION
**Pull request by bernd-wechner**
This proved a little bigger than hoped, but not too bad. As noted earlier, `webdb.py` and `releasewidget.py` were out of kilter with the rest of the UI definition which rests under `mainwin` So I moved them into `mainwin` and refactored. It is the refactoring that grew a little. To wit, there are small incremental commits along the way.

1. Move the two source files and refactor all code references. Test run, and all good.
2. The translation files contained <location> references to the old source files, refactored these. Waste of time as later they are all rebuilt anyhow in 5 below.
3. Merged the WebdDB translation context into the Tag Sources context (which was tiny) and the look and feel is much better in QT Linguist. Re-released them all after the refactor, again a waste of time as they all get updated later in 5 below.
4. Test `update_translation.py` and find it broken. Fixed the code so it works.
5. Running it though updated all the `.ts` files with new location refs (having performed a code scan), so released them all as well using `lrelease` to update the `.qm` files.
6. Finally, removed some dated python2 refs in the translation doc.

It looks larger than it is, with most of the resulting file changes to `.ts` and `.qm` files thanks to `lupdate` and `lrelease`.

The upshot is now, slightly cleaner code, with the Tagsources window definition more sensibly named and located, and the translation files all with updated locations and re-released.

### CODE REVIEW

**Comment by corubba** [comment]
Since you moved sources, you will need to update `SOURCES` in `puddletag.pro` for `update_translation.py` to work properly. Otherwise it throws (non-fatal) errors:
```
pylupdate5 error: Cannot open Python source file 'puddlestuff/releasewidget.py': No such file or directory
pylupdate5 error: Cannot open Python source file 'puddlestuff/webdb.py': No such file or directory
```

Looks fine otherwise. Will be fun to re-integrate #687 with this.

**Comment by bernd-wechner** [comment]
Will do! Thanks for the heads up!

**Comment by bernd-wechner** [comment]
Can we get a move on here? I see a lot of activity elsewhere, and I want to see this merged, so I can get cracking on A Discogs Tagsource. I had a long break as we all came down with COVID and my workstation went down in sympathy and needed a rebuild ;-). But I'd like to know what needs doing here, and I'll look into it.

**Comment by sandrotosi** [comment]
@bernd-wechner we do welcome every contribution, but we do not welcome un-necessary pressure: we are all volunteers, that we work on what we like when we have time.

As for what you need to do to move this forward: this PR has conflicts on 4 files; please fix that, otherwise it cant be merged and so i wont be looking at anything else so far.

thanks

**Comment by bernd-wechner** [comment]
Thanks.  Been very busy on numerous other commitments.  But this is on the agenda some time.

**Comment by bernd-wechner** [comment]
How's that? I haven't inspected closely. Between many things, I pulled master from usptream and then to local, then rebased this branch onto master and resolved all merge conflicts (in favour of Master as it happens), then pushed the branch again to Remote (forced). And now I see here (I'm not quite sure of the exact magic of github) this is reporting as expected "This branch has no conflicts with the base branch". So I'm hoping it's good.

In time, I need to turn back to the tags sources work I was on and fixing it so I can continue tagging my library too (all fell by the wayside). The activity I've seen here via notifications has motivated me a little to squeeze some time in ;-). And this change is sort of central to sensible progress on tagsources methinks (though that is subjective of course)

**Comment by bernd-wechner** [comment]
Do consider merging this or feeding back before more merge conflicts arise because Master diverges. I'll also wait for this merge before I put any more energy tag sources. Not pressuring, I know what FOSS is like, and heck not like I've been here much in the past year (and have a load of PRs that need my attention elsewhere too - it's all about priorities and distractions). Just making my interest clear, is all.

**Comment by bernd-wechner** [comment]
Loving all the activity here of late. Fingers crossed this little change gets some love ;-)

**Comment by sandrotosi** [comment]
i would still like an extra review from @corubba but all in all it looks fine to me, and it even fixes some of the issues i was having with #788 ! So if there are not additional changes, i think we can be ready to merge it

im not a big fan of code changes/reorgs interleaved with reformatting (f.e. all the extra empty lines in the first commit are quite distracting; and while i understand it could be an automation of the IDE used, they belong in a separate/independent commit)

**Comment by bernd-wechner** [comment]
Sorry if my IDE added unwelcome whitespace. Let me know which file(s) affected and I'll see if I can tidy it up if needed. I concur, format/layout changes are distracting from code changes. Thanks heaps for finally checking it out.

**Comment by sandrotosi** [comment]
> Sorry if my IDE added unwelcome whitespace. Let me know which file(s) affected and I'll see if I can tidy it up if needed. I concur, format/layout changes are distracting from code changes. Thanks heaps for finally checking it out.

naah it's ok as is, just a gentle reminder for next time

File: docs/_sources/source/translate.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ Then run update_translation.py with your desired locale (eg en_ZA, rus, fr) like
 
 .. code-block:: bash
 
-    python2 update_translation.py your_locale
+    python update_translation.py your_locale
```
[the remark this answers is not in the document]
Comment (bernd-wechner) [reply docs/_sources/source/translate.txt:24]: Agreed. I had assumed that all distros had moved to python3 as default. I mean python2 was deprecated 3 years ago now and Python3 is already on release 3.11. But of course I don't object to is reading python3 if we prefer. That's cool too of course. Just hoy if you'd like me to update the branch. I'm just waiting for a merge now so I can pull and merge master and continue on tagsources ... (at leisure, so also not in any hurry, we're all busy ;-)

[the remark this answers is not in the document]
Comment (sandrotosi) [reply docs/_sources/source/translate.txt:24]: i think it's true that all modern distros have moved to python3, not all of them made `python` point to `python3` (i think Arch was the first one to do so); f.e. on Debian-based distro, there's a compatibility package called `python-is-python2` (to make `python -> python2` symlink) and python-is-python3` (to make `python -> python3` symlink) that you have to manually install if you want to have it manage what `/usr/bin/python` means on your machine

**Comment by sandrotosi** [comment]
at long last, this is merged :) thanks @bernd-wechner for your contribution!

**Comment by bernd-wechner** [comment]
Thanks to you all for the top product. Apologies for seriously slow responses mysides too (I have my finger in way too many pies I think and am here now because believe it or not am closing old browser tabs ;-). As time permits I'll look again at music and at the discogs cover interface in puddletag (as I had that underway too at some point).

**Pull request merged**